scalyr

```diff
@@ -55,6 +56,33 @@ def getApiToken(args, environmentVariableName, permissionType):
     return apiToken
 
 
+def getProxyAddress(args):
+    proxyAddress = args.proxy
+
+
+    if proxyAddress is None:
+       #Check if we have a environment variable
+       if os.environ.get('http_proxy'):
```

Comment:
Should add a `scalyr_` prefix to the environment variable names to help avoid possible conflicts with other applications.

Reply:
I think I was trying to use the default Linux proxy systems, like a requests would.

- http_proxy
- https_proxy
- HTTP_PROXY
- HTTPS_PROXY

Hmm, this looks like a cleaner implementation and it would be using stdlib:

https://github.com/psf/requests/blob/4f6c0187150af09d085c03096504934eb91c7a9e/requests/compat.py#L38 

https://github.com/psf/requests/blob/4f6c0187150af09d085c03096504934eb91c7a9e/requests/compat.py#L59

Reply:
Python3: https://docs.python.org/3/library/urllib.request.html#urllib.request.getproxies
Python2: https://docs.python.org/2.7/library/urllib.html#urllib.getproxies

Reply:
You're right, this approach is good.